doc/neptune-metadata.rst

```diff
@@ -0,0 +1,109 @@
+===========================
+Neptune Metadata Definition
+===========================
+
+This file specifies what fields are in the ``.neptune.yml`` file, what they mean
+and by which tools they are used. The file itself is expected to be in a
+repository's root directory.
+
+.. contents::
+
+library
+-------
+
+Type
+  boolean
+Meaning
+  If set to true, the repository is marked as a library used by other repositories
+Tools
+  - neptune-overview
+  - neptune-autopr
+Example
+  .. code:: yml
+
+    library: true
+
+d2ready
+-------
+
+Type
+  String
+Meaning
+  Can have one of the following values:
+  - ``false``: No D2 support whatsoever (default)
+  - ``true``: synonym for ``convertable``
+  - ``convertable``: repository can automatically be converted to D2
+  - ``only``: only D2 is supported
+Tools:
+  - neptune-overview
+Example
+  .. code:: yml
+
+    d2ready: true
+
+support-guarantees
+------------------
+
+Type
+  Mapping
+Meaning
+  Sets the support window and period users can expect for this repository
+Tools:
+  - neptune-overview
+  - neptune-autopr
+Example
+  .. code:: yml
+
+    support-guarantees:
+      major-months: 6
+      minor-versions: 2
+
+major-months
+~~~~~~~~~~~~
+
+Type
+  Integer
+Meaning
+  Amount of months a major version is supported, starting from the time the next
+  major version is released
+
+minor-versions
+~~~~~~~~~~~~~~
+
+Type
+  Integer
+Meaning
+  Amount of minor versions that are supported, starting from the latest one
+  
+automatic-update-prs-global
+---------------------------
+
+Type
+  String
+Meaning
+  Defines which types of auto-pull-requests the repository desires. Possible
+  values are "None", "Patch", "Minor", "Major"
```

Comment:
Why the capitalization? Since everything is lowercase so far I would use capitalized words here (or make it case insensitive and specify it is case insensitive). But I would rather have it all in lowercase to avoid complexity.

Reply:
Well. I guess because I treat them as constants and in my mind constants are capitalized